client/src/overview/subComponents/ImageGallery.jsx

```diff
@@ -1,9 +1,50 @@
-import React from 'react';
+import React, { useContext, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { StyleIdxContext } from '../overview.jsx';
+
+export default function ImageGallery({ currentProductStyles }) {
+  const StyleIdxContextData = useContext(StyleIdxContext);
+  const stylePics = [];
+  const [currentSlide, setCurrentSlide] = useState(0);
+
+  let slideLength = 0;
+  let currentStyleIdx;
+  if (StyleIdxContextData && currentProductStyles) {
+    ({ currentStyleIdx } = StyleIdxContextData);
+    currentProductStyles.results[currentStyleIdx].photos.forEach((photo, index) => {
+      stylePics.push(photo.url);
+    });
+    slideLength = stylePics.length;
+  }
+
+  const changeSlide = (event, handle) => {
+    if (handle === 'next') {
+      setCurrentSlide(currentSlide === slideLength - 1 ? 0 : currentSlide + 1);
+    } else if (handle === 'prev') {
+      setCurrentSlide(currentSlide === 0 ? slideLength - 1 : currentSlide - 1);
+    }
+  };
+
+  useEffect(() => {
+    setCurrentSlide(0);
+  }, [currentStyleIdx]);
 
-export default function ImageGallery() {
   return (
     <div className='imageGallery'>
-      <h3>Image Gallery!</h3>
+      <button className={ 'fas fa-chevron-left left-arrow arrow'} onClick={ () => changeSlide(event, 'prev') } />
+      <button className={ 'fas fa-chevron-right right-arrow arrow' } onClick={ () => changeSlide(event, 'next') } />
+      {stylePics.map((stylePic, index) => {
+        return (
+          <div key={index} className={index === currentSlide ? 'slide-active' : 'slide'}>
```

Comment:
Generally don't want to use only an index as the key, is there any unique prop on the obj that can be utilized. Maybe the index and a slice of a string from the obj if there isn't a unique id on it?